pkgs/os-specific/linux/kernel/perf/default.nix

```diff
@@ -143,7 +143,8 @@ stdenv.mkDerivation {
   ++ lib.optional withZstd zstd
   ++ lib.optional withLibcap libcap
   ++ lib.optional (lib.versionAtLeast kernel.version "5.8") libpfm
-  ++ lib.optional (lib.versionAtLeast kernel.version "6.0") python3.pkgs.setuptools;
+  ++ lib.optional (lib.versionAtLeast kernel.version "6.0") python3.pkgs.setuptools
+  ++ lib.optional (lib.versionAtLeast python3.version "3.12") python3.pkgs.distutils; # Python 3.12 no longer includes distutils
```

Comment:
Please add `lib.versionOlder kernel.version "6.0"` as a predicate, as newer kernels don't need it any more.